src/lib.rs

```diff
@@ -505,9 +502,13 @@ where
     /// Run a temperature/humidity measurement and return the combined result.
     ///
     /// This is a blocking function call.
-    pub fn measure(&mut self, mode: PowerMode) -> Result<Measurement, Error<E>> {
+    pub fn measure<D: DelayUs<u16>>(
+        &mut self,
+        mode: PowerMode,
+        delay: &mut D,
+    ) -> Result<Measurement, Error<E>> {
```

Comment:
You could also use the impl trait syntax:

```suggestion
    pub fn measure(
        &mut self,
        mode: PowerMode,
        delay: &mut impl DelayUs<u16>,
    ) -> Result<Measurement, Error<E>> {
```

Might be easier to read, what do you think?

Reply:
Fixed.